src/core/bpf-firewall.c

```diff
@@ -592,13 +662,42 @@ int bpf_firewall_install(Unit *u) {
         }
 
         if (u->ip_bpf_ingress) {
-                r = bpf_program_cgroup_attach(u->ip_bpf_ingress, BPF_CGROUP_INET_INGRESS, path, flags);
+                r = bpf_program_cgroup_attach(u->ip_bpf_ingress, BPF_CGROUP_INET_INGRESS, path,
+                                              u->ip_bpf_custom_ingress ? BPF_F_ALLOW_MULTI : flags);
                 if (r < 0)
                         return log_unit_error_errno(u, r, "Attaching ingress BPF program to cgroup %s failed: %m", path);
 
                 u->ip_bpf_ingress_installed = bpf_program_ref(u->ip_bpf_ingress);
         }
 
+        SET_FOREACH(prog, u->ip_bpf_custom_egress, i) {
+                r = bpf_program_cgroup_attach(prog, BPF_CGROUP_INET_EGRESS, path, BPF_F_ALLOW_MULTI);
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Attaching custom egress BPF program to cgroup %s failed: %m", path);
+                /* Remember that these BPF programs are installed now. */
+                r = set_ensure_allocated(&u->ip_bpf_custom_egress_installed, NULL);
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Can't allocate BPF program set: %m");
+
+                r = set_put(u->ip_bpf_custom_egress_installed, bpf_program_ref(prog));
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Can't add program to BPF program set: %m");
+        }
+
+        SET_FOREACH(prog, u->ip_bpf_custom_ingress, i) {
+                r = bpf_program_cgroup_attach(prog, BPF_CGROUP_INET_INGRESS, path, BPF_F_ALLOW_MULTI);
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Attaching custom ingress BPF program to cgroup %s failed: %m", path);
+                /* Remember that these BPF programs are installed now. */
+                r = set_ensure_allocated(&u->ip_bpf_custom_ingress_installed, NULL);
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Can't allocate BPF program set: %m");
+
+                r = set_put(u->ip_bpf_custom_ingress_installed, bpf_program_ref(prog));
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Can't add program to BPF program set: %m");
+        }
```

Comment:
it appears to me it would be relatively nice and easy to place this loop into a function that an then be called once for ingress and once for egress